pkg/render/prometheus_service.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"net/url"
+	"strconv"
+
+	operator "github.com/tigera/operator/api/v1"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	calicoNodePrometheusServiceName = "calico-node-prometheus"
+
+	tigeraPrometheusServiceName         = "tigera-prometheus-service"
+	prometheusEndpointUrlEnvVarName     = "PROMETHEUS_ENDPOINT_URL"
+	prometheusOperatedHttpServiceScheme = "http"
+	prometheusOperatedHttpServiceHost   = "prometheus-operated-http.tigera-prometheus"
+
+	prometheusServiceListenAddrEnvVarName = "LISTEN_ADDR"
+
+	tigeraPrometheusServiceHealthEndpoint = "/health"
+)
+
+func TigeraPrometheusService(cr *operator.InstallationSpec, pullSecrets []*corev1.Secret, prometheusServicePort int) Component {
```

Comment:
calling this PrometheusService is a tad confusing to me, since a `Service` is a specific kind of thing in Kubernetes, and this renderer creates more than just a `Service` (for example, it also has a `Deployment`).

Reply:
yeah sorry naming this was hard, using `Service` because the proxy might potentially evolve to something  closer to a full fledged api service, maybe i'll change to naming it to `tigera-prometheus-api`

Reply:
https://github.com/tigera/operator/pull/1379/commits/4ec097420c2e67725222b5db380aecff96ca1727